ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -286,23 +298,26 @@ IndexValue find(StoreKey keyToFind) throws StoreException {
             if (result == 0) {
               byte[] buf = new byte[valueSize];
               duplicate.get(buf);
-              return new IndexValue(startOffset.getName(), ByteBuffer.wrap(buf));
+              toReturn = new IndexValue(startOffset.getName(), ByteBuffer.wrap(buf));
+              break;
             } else if (result < 0) {
               low = mid + 1;
             } else {
               high = mid - 1;
             }
           }
-          metrics.bloomFalsePositiveCount.inc(1);
+          if (toReturn == null && bloomPositive) {
```

Comment:
Nice. I think we can even get rid of this variable. Could you try to refactor this as follows:
```java
if (bloomFilter != null && !bloomFilter.isKeyPresent()) {
  // negative
} else {
  // search
  if (bloomFilter != null && toReturn == null) {
    // false positive
  }
}
```
Basically `bloomPositive == bloomFilter != null` once you are in the block where you do the binary search.